lib/arborist/build-ideal-tree.js

```diff
@@ -693,9 +695,11 @@ module.exports = cls => class IdealTreeBuilder extends cls {
   // where this dep cannot be placed, and use the one right before that.
   // place dep, requested by node, to satisfy edge
   [_placeDep] (dep, node, edge, peerEntryEdge = null) {
-    if (edge.to && !edge.error && !this[_updateNames].includes(edge.name) &&
-        !this[_isVulnerable](edge.to))
-      return []
+    if (edge.to &&
+      !edge.error &&
+      !this[_updateNames].includes(edge.name) &&
+      !this[_isVulnerable](edge.to) &&
+      !this[_dedupeList].has(edge.to.path)) return []
```

Comment:
style nit, should be:

```js

if (edge.to &&
    !edge.error &&
    !this[_updateNames].includes(edge.name) &&
    !this[_isVulnerable](edge.to) &&
    !this[_dedupeList].has(edge.to.path))
  return []
```

- Double-indent for line-breaks within a `()`
- Always a line-break for single-line conditionals